src/workflow-engine/charms/temporal-worker/README.md

```diff
@@ -4,138 +4,184 @@ This folder provides the Temporal Workflows and Activities for the existing
 
-### Additional prerequisites
+## Deploying the charm with Terraform
 
-- `terraform` (>=v1.5.5)
-- this TF configuration is designed to use the MicroK8s cluster deployed
-[here](../../README.md). Before proceeding, make sure you:
-  - connect to the VPN,
-  - forward all the necessary remote endpoints to your `localhost``, with:
+This is the preferred option (Terraform should already be installed, but if not,
+run `sudo snap install terraform --classic`).
 
-      ```bash
-      ssh -L 5000:<keystoneUrl>:<keystonePort> -L ...<username>@<internalOSServer>
-      ```
+### Additional prerequisites
 
-    this is needed because of Bastion,
-- (optional) if you are managing an already existing deployment, **you need**
+- **if** you are managing an already existing deployment, **you need**
 to request and copy the corresponding `terraform.tfstate` file into this
 directory. NOTE: this file may contain sensitive information so please do not
 commit or share it.
 
-`terraform apply`
+### Deploy
 
-TIP: create a `variables.tfvars` key-value file with your test input parameters
-defined so you don't always have to pass them to `terraform apply`, using
-instead `terraform apply -var-file="variables.tfvars"`. Don't worry, this file
-is ignored by Git (see `.gitignore`).
+> TIP: create a `variables.tfvars` key-value file with your test input
+> parameters
+> defined so you don't always have to pass them to `terraform apply`, using
+> instead `terraform apply -var-file="variables.tfvars"`. Don't worry, this file
+> is ignored by Git (see `.gitignore`).
 
-### Troubleshooting
+Start by running:
 
-> │ Error: dial tcp 10.152.183.75:17070: i/o timeout  
-> │  
-> │   with provider["registry.terraform.io/juju/juju"],  
-> │   on charm.tf line 32, in provider "juju":  
-> │   32: provider "juju" {}  
-> │  
-> │ Connection error, please check the controller_addresses property set on the
-provider
+```bash
+terraform init
+```
 
-If this happens, you're likely running Terraform from a host outside the
-MicroK8s cluster. In this case, you'll want to double-check your Juju
-controller's network configuration, making sure it is being exposed to an
-external IP your host can reach. See [how to configure the controller](#external-ip) above.
+In `charm.tf` you'll find two different settings for the Juju provider:
+
+**JUJU SETTINGS 1**: relies on the Juju controller having already been
+configured/used in the local machine where you are running Terraform (e.g. as
+shown below in the "[From your workstation](#from-your-workstation)" section).
+I.e.:
+
+  1. the following environment variables, that correspond to the configuration
+  fields indicated above, are present:
+     - JUJU_CONTROLLER_ADDRESSES
+     - JUJU_USERNAME
+     - JUJU_PASSWORD
+     - JUJU_CA_CERT
+
+     or
+
+  1. you have used this Juju controller locally before. This is the most straightforward solution. Remember, that it will use the configuration
```

Comment:
it's 2. not 1. ?

Reply:
that doesn't matter in MD. See https://github.com/canonical/oci-factory/pull/126/files/9740ec4628c4feb02090f6d27e01134a993d60fd?short_path=bc261af#diff-bc261af0e121c07f8d041b0a8586f0e73da2d0ec5faff7b2bffc5f0720034fd1

Changing anyway

Reply:
https://github.com/canonical/oci-factory/pull/126/commits/c919524c1f82370712482dcd2a204d82198d1069